agentstack/generation/tool_generation.py

```diff
@@ -78,6 +79,8 @@ def remove_tool(tool_name: str, path: Optional[str] = None):
             os.remove(f'{path}src/tools/{tool_name}_tool.py')
             remove_tool_from_tools_init(tool_data, path)
             remove_tool_from_agent_definition(framework, tool_data, path)
+            if tool_data.get('post_remove'):
```

Comment:
good thinking!

Reply:
I think in general we should try to avoid the install/remove patter in the configs and try our best to handle it by providing additional context like we do with `packages` now. But, there's always an edge case, so having explicit, optional hooks is a good thing, too.